### DISCUSSION TITLE
Language selection

### DISCUSSION THREAD

**Discussion by ErinvanderVeen** [topiary/topiary#561] [Ideas]
From #447 

> Topiary is expected to be multi language, and the number of supported languages will surely grow. In that situation, it can happen that you want to use Topiary e.g. only for one or a restricted number of languages but use other formatters for other languages. Typically, topiary is the only formatter able to handle Nickel right now, but I don't want it to try to format the bash scripts of the Nickel repo.

I argue that `topiary -f **/*` will be a reasonably common usecase. However, one might not want to use Topiary for EVERY language.

Topiary can already be configured to change languages using the configuration file (like indentation and extensions), but the only way to DISABLE a language right now (given the usecase above) would be overriding the extensions field to the empty list. Is that acceptable?

Are there better ways for us to support that usecase?

**Comment by ErinvanderVeen** [comment]
Alternatively a parameter like `--languages` to allowlist a subset of supported languages.

**Reply by Xophmeister** [reply]
It's quite common for CLI tools to have flags for either whitelisting or blacklisting, whenever processing multiple files, so I think this seems like a reasonable approach.

**Comment by BirdeeHub** [comment]
In the meantime, I have been using [treefmt](https://github.com/numtide/treefmt) for this. But I am also in favor of a whitelisting behavior, and it would match up nicely with the treefmt options as well.
